relay.go

```diff
@@ -302,6 +310,32 @@ func (r *relay) mailHandler(cfg *config) func(ctx context.Context, peer smtpd.Pe
 
 		deliveryLog.InfoContext(ctx, "delivering mail from peer using smarthost")
 
+		// successful status is always 250
+		statusCode := 250
+		start := time.Now()
+
+		defer func() {
+			span.SetAttributes(traceutil.StatusCode(statusCode))
+
+			observeDuration(ctx, statusCode, time.Since(start))
+		}()
+
+		// apply rate limiting if enabled
+		if r.rateLimiter != nil {
+			sender := env.Sender
+			switch {
+			case sender == "":
+				// allow requests with empty sender
+				logger.WarnContext(ctx, "empty sender address for rate limiting, allowing by default")
+			case !r.rateLimiter.allow(sender):
+				logger.WarnContext(ctx, "rate limit exceeded", slog.String("sender", sender))
+
+				statusCode = smtpd.ErrRateLimitExceeded.Code
+
+				return observeErr(ctx, smtpd.ErrRateLimitExceeded)
+			}
+		}
```

Comment:
Want to get some feedback on this approach...

Right now limits are applied per-sender vs per-slug. It did not seem obvious how we could reliably get the slug in the context of this smtprelay service, especially considering custom domains. So this approach seemed like it might simplify things and achieve a similar end result without needing to make this relay "slug-aware". My only concern is around the reliability of the `env.Sender` field, which presumably comes from the `MAIL FROM` field in SMTP, which can be spoofed easily. We could alternatively do it per-IP address, but not sure if that would work for us or not.